### DISCUSSION TITLE
Summaries - Habits and Summaries - Averages not correct

### DISCUSSION THREAD

**Discussion by chrismalek** [NotePlan/plugins#443] [Q&A]
I started to use Summaries last month and just completed a few weeks of some items I want to track. Overall this is really good plugin. It took me a few years to really understand it after hitting the readme a few times.

I am committed now.

Anyways there is something not correct with the averages.

I log my daily weight like this:

`@weight(197.5).`

When I run my stats for this month I get this:

`        weight |···············███████████████|` avg 200 (from 15)

However, None of the days I logged were above 198. So why does it average to 200?

**Comment by jgclark** [comment]
@chrismalek I've got one idea about this: I've set it to give averages to 2 significant figures. This avoids spurious accuracy on most measurements (e.g. was max heart rate really 178 and not 177? Did I really sleep 6.7 or 6.8 hours?). But in the case of weight, I can see that there might be more accuracy available.  Can you give me a sample of 10 actual readings, so I can work out a better way of displaying the averages?

**Comment by chrismalek** [comment]
Here are all my current weight values from the side bar.

<img width="308" alt="image" src="https://github.com/NotePlan/plugins/assets/9403/e434bcc4-7072-4ca0-916e-a996b21ba54f">



I only have one entry that is actually 200 and the rest are around 197 or below.

My current monthly period stat for that looks like this but I know my average should be really around 197 or 198.  

`        weight |██████████|` avg 200 (from 10)

**Reply by jgclark** [reply]
OK, I will change this to use 3 significant figures, not 2, and then the average is what you expect. I'm on holiday at the moment, so I probably won't get to release this for a few days.

**Comment by chrismalek** [comment]
@jgclark  tThanks enjoy your holiday!   :-)

**Comment by chrismalek** [comment]
@jgclark I just updated to v0.19.0 and I see the averages. However, the /periodStats command no longer produces any output. The appendProgressUpdate command does work. 
I need to look through the docs to figure out how to turn on debugging for these plugins. I think I saw it mentioned in past reading.

**Reply by jgclark** [reply]
Bother. I did a lot of refactoring in the last update, and something clearly slipped passed me.
I can reproduce it, so don't worry about sending any logs.

**Comment by chrismalek** [comment]
Here is the log console. It might be something I need to adjust now in my settings for this.

```log
Executing function 'statsPeriod'
2023-05-14 08:49:34 | INFO  | jgclark.Summaries v0.19.0 :: statsPeriod: starting for May 2023 (2023-05-01 - 2023-05-14)
2023-05-14 08:49:34 | ERROR | gatherOccurrences :: undefined is not an object (evaluating 'a.progressYesNo')
Note opened notification, filename = 2023-05.md, waiting for: 2023-05.md, matches = true

```
